libs/openai-assistant/src/lib/assistant/assistant-files.service.ts

```diff
@@ -3,29 +3,103 @@ import { FileObject } from 'openai/resources';
 import { createReadStream } from 'fs';
 import { AiService } from '../ai';
 import { ConfigService } from '../config';
-
+import { AssistantCreateParams, VectorStore } from 'openai/resources/beta';
+import ToolResources = AssistantCreateParams.ToolResources;
+import { AssistantUpdateParams } from 'openai/src/resources/beta/assistants';
+import {
+  AssistantCodeInterpreter,
+  AssistantFileSearch,
+  AssistantToolResources,
+} from './assistant.model';
 @Injectable()
 export class AssistantFilesService {
   constructor(
     private readonly assistantConfig: ConfigService,
     private readonly aiService: AiService,
   ) {}
 
-  async create(
-    fileNames: string[],
+  async getFiles(
+    attachments: string[] = [],
     fileDir = this.assistantConfig.get().filesDir,
-  ): Promise<string[]> {
+  ): Promise<FileObject[]> {
     const files: FileObject[] = [];
 
-    for (const name of fileNames) {
-      const file = await this.aiService.provider.files.create({
-        file: createReadStream(`${fileDir || ''}/${name}`),
-        purpose: 'assistants',
-      });
+    await Promise.all(
```

Comment:
It could be beneficial considering to add error handling. Currently, if one of the file creations or operations fails, it could reject the whole Promise.

Reply:
Or maybe there is a concept I did not catch

Reply:
I've added the error handler for a parent method